app/src/main/java/com/xinto/mauth/ui/viewmodel/MainViewModel.kt

```diff
@@ -1,12 +1,59 @@
 package com.xinto.mauth.ui.viewmodel
 
+import android.content.Intent
+import android.os.Bundle
+import androidx.compose.runtime.State
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.xinto.mauth.domain.model.DomainAccountInfo
 import com.xinto.mauth.domain.repository.MainRepository
+import com.xinto.mauth.otp.parser.OtpUriParser
+import com.xinto.mauth.otp.parser.OtpUriParserResult
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
 
 class MainViewModel(
-    mainRepository: MainRepository
+    mainRepository: MainRepository,
+    private val optParser: OtpUriParser,
 ) : ViewModel() {
 
     val privateMode = mainRepository.observeSecureMode()
 
+    private val _optUri = mutableStateOf<DomainAccountInfo?>(null)
```

Comment:
```suggestion
    var optUri = mutableStateOf<DomainAccountInfo?>(null)
        private set
``` is more readable since one variable now does both things. update the .value calls accordingly (`otpUri.value = something` becomes `otpUri = something`